pkgs/matcher/lib/src/core_matchers.dart

```diff
@@ -148,9 +148,9 @@ class _ReturnsNormally extends FeatureMatcher<Function> {
   const _ReturnsNormally();
 
   @override
-  bool typedMatches(Function f, Map matchState) {
+  bool typedMatches(Object f, Map matchState) {
```

Comment:
Is this better? Better to ignore?

Reply:
Either pattern is ugly.

We'd either need to ignore an `unnecessary_cast` or the `avoid_dynamic_calls` if we use an ignore.

Do we think it would be better behavior in the analyzer to suppress `unnecessary_cast` in these cases?
If we think the analyzer would make that change adding an `// ignore: unnecessary_cast` would be a good choice for now.

Reply:
https://github.com/dart-lang/sdk/issues/56939

I'll ignore the `unnecessary_cast` for now and if the discussion on that issue changes we can rethink which pattern to use here.